sdk/metric/instrumentkind.go

```diff
@@ -0,0 +1,34 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.17
+// +build go1.17
+
+package metric // import "go.opentelemetry.io/otel/sdk/metric"
+
+// InstrumentKind describes the a kind of instrument.
+type InstrumentKind uint8
+
+// The following represent the different kind of instruments that can be
+// created by the SDK.
+const (
+	//nolint:deadcode,varcheck
+	undefinedInstrument InstrumentKind = iota
+	SyncCounter
+	SyncUpDownCounter
+	SyncHistogram
+	AsyncCounter
+	AsyncUpDownCounter
+	AsyncGauge
+)
```

Comment:
This reads weird on the docs site where this comment comes below the enum listing.

![20220609_094823](https://user-images.githubusercontent.com/5543599/172901245-d259d5d9-41a7-4e07-9f7c-4ea6a5e411a9.png)

```suggestion
// These are all the instrument kinds supported by the SDK.
const (
	// undefinedInstrument is an uninitialized instrument kind, should not be used.
	//nolint:deadcode,varcheck
	undefinedInstrument InstrumentKind = iota

	// SyncCounter is an instrument kind that records increasing values
	// synchronously in application code.
	SyncCounter
	// SyncUpDownCounter is an instrument kind that records increasing and
	// decreasing values synchronously in application code.
	SyncUpDownCounter
	// SyncHistogram is an instrument kind that records a distribution of
	// values synchronously in application code.
	SyncHistogram
	// AsyncCounter is an instrument kind that records increasing values in an
	// asynchronous callback.
	AsyncCounter
	// AsyncUpDownCounter is an instrument kind that records increasing and
	// decreasing values in an asynchronous callback.
	AsyncUpDownCounter
	// AsyncGauge is an instrument kind that records current values in an
	// asynchronous callback.
	AsyncGauge
)
```

This will give these package docs:

![20220609_115943](https://user-images.githubusercontent.com/5543599/172923881-a9bce66c-a7f4-4ecf-adc8-e61b780ac27b.png)